### ISSUE TITLE
NPE in PMD Preferences updating to 4.0.18.v20181220-1448

### ISSUE THREAD

**Issue by KristofSzabados** [pmd/pmd-eclipse-plugin#74]
Since updating when I try to visit the Rule Configuration preference page I get these 2 errors:

```
java.lang.NullPointerException
	at net.sourceforge.pmd.eclipse.ui.preferences.br.PMDPreferencePage2.saveUIState(PMDPreferencePage2.java:537)
	at net.sourceforge.pmd.eclipse.ui.preferences.br.PMDPreferencePage2.performCancel(PMDPreferencePage2.java:502)
	at org.eclipse.jface.preference.PreferenceDialog$1.run(PreferenceDialog.java:266)
...
```

and 

```
java.lang.UnsupportedOperationException
	at java.util.Collections$UnmodifiableMap.remove(Unknown Source)
	at net.sourceforge.pmd.eclipse.ui.preferences.panelmanagers.Configuration.filteredPropertiesOf(Configuration.java:29)
	at net.sourceforge.pmd.eclipse.ui.preferences.br.RuleUIUtil.indexedPropertyStringFrom(RuleUIUtil.java:160)
	at net.sourceforge.pmd.eclipse.ui.preferences.br.ImageColumnDescriptor$1.valueFor(ImageColumnDescriptor.java:28)
	at net.sourceforge.pmd.eclipse.util.AbstractCellPainterBuilder.valueFor(AbstractCellPainterBuilder.java:97)
	at net.sourceforge.pmd.eclipse.util.StyledTextBuilder.adjust(StyledTextBuilder.java:35)
	at net.sourceforge.pmd.eclipse.util.StyledTextBuilder.access$0(StyledTextBuilder.java:33)
	at net.sourceforge.pmd.eclipse.util.StyledTextBuilder$2.handleEvent(StyledTextBuilder.java:81)
	at org.eclipse.swt.widgets.EventTable.sendEvent(EventTable.java:84)
...
```

* and a dialog appears telling me it could not accept changes.

**Comment by adangel** [comment]
Hi @KristofSzabados ,

the second exception (UnsupportedOperationException) is fixed in the current snapshot (see #70 and #75). The NPE I can't reproduce. Could you update to the latest snapshot via update site https://dl.bintray.com/pmd/pmd-eclipse-plugin/snapshots/updates/ and test it again?

**Comment by KristofSzabados** [comment]
This looks like fixed in 4.1.0
